src/vue-property-decorator.ts

```diff
@@ -26,13 +26,36 @@ export function Inject(key?: string | symbol): PropertyDecorator {
 }
 
 /**
+ * decorator of a provide
+ * @param key key
+ * @return PropertyDecorator | void
+ */
+export function Provide(key?: string | symbol): PropertyDecorator {
+  return createDecorator((componentOptions, k) => {
+		var provide: any = componentOptions.provide;
+		if('function'!== typeof provide || !provide.managed) {
```

Comment:
`managed` is the list of the "managed provides" (the ones managed by this library) - it is a map {k: v} where `v` is the name used in the "provide" list, and `k` is the property of the instance that is queried when providing.
Everything is a string here - I don't see a use for a WeakMap